### PR TITLE
Add new text format for `istore` and new row conversion methods.

### PR DESCRIPTION
**Pull request by ildus**
Adds text format like: '([1,2,3], [11,22,33])'. It can be described as tuple of two arrays.
Also adds two new functions for `istore` construction: `row_to_istore`, `row_to_bigistore`. These functions get `record` type which should contain two arrays.

All of this is going to be used in `clickhouse_fdw` to make implicit conversion of `clickhouse` tuple to `istore` column.

### CODE REVIEW

**Comment by mkabilov** [comment]
@rapimo can you have a look at this PR?

File: test/expected/istore_io_test.out [hunk 1]
```diff
@@ -144,6 +144,122 @@ LINE 1: SELECT '2=>4, 10=5, 5=>17'::bigistore;
...
+----------------------
+ "1"=>"11", "2"=>"22"
+(1 row)
+
+ROLLBACK;
+BEGIN;
+-- bigistore istore_io array input should parse uneven arrays tuple;
+-- ./spec/istore/istore_io_spec.rb:86;
+CREATE EXTENSION istore;
+SELECT '([1,2,2], [11, 22])'::bigistore;
```
[the remark this answers is not in the document]
Comment (ildus) [reply test/expected/istore_io_test.out]: No special reason, just when it's going through text if one of arrays is suddenly finished it counts it as we're done with the pairs. I didn't expect this format to be documented and used by anyone except from us, so have not added any checks in this case.
But that's a valid point, I will add this checks.

[the remark this answers is not in the document]
Comment (ildus) [reply test/expected/istore_io_test.out]: fixed

File: spec/istore/istore_io_spec.rb [hunk 2]
```diff
@@ -77,23 +77,22 @@
...
         end
       end
-
       describe 'array input' do
         it 'should parse normal arrays tuple' do
           query("SELECT '([1,2], [11, 22])'::#{type}").should match \
           '"1"=>"11", "2"=>"22"'
         end
-        it 'should parse uneven arrays tuple' do
-          query("SELECT '([1,2,2], [11, 22])'::#{type}").should match \
```
Comment (rapimo) [inline spec/istore/istore_io_spec.rb]: I would prefer to keep the test just change the expectation to raise

Comment (ildus) [reply spec/istore/istore_io_spec.rb]: yes, I did that in latest commits

**Comment by ildus** [comment]
I have fixed all review comments, merging

**Pull request merged**